operations/kubernetes/csm-service-upgrade-FAILED.md

```diff
@@ -0,0 +1,193 @@
+services            cray-dns-unbound-manager-1650373560-wsnl5                         0/2     Completed          0          16m
+services            cray-dns-unbound-manager-1650374520-mq4wk                         0/2     Completed          0          4m57s
+services            cray-dns-unbound-manager-1650374520-tdhsl                         1/2     Error              0          2m14s
+```
+
+Continuing the troubleshootng and looking for most recent log we can understand that it can't talk to Kea.
+
+```sh
+ncn-m001:~ # kubectl -n services logs cray-dns-unbound-manager-1650374520-tdhsl -c manager
+2022-04-19 13:24:53,530 - __main__ - INFO - Querying Kea in the cluster to find any updated records we need to set
+Traceback (most recent call last):
+  File "/usr/lib/python3.8/site-packages/requests/adapters.py", line 439, in send
+    resp = conn.urlopen(
+  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 846, in urlopen
+    return self.urlopen(
+  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 846, in urlopen
+    return self.urlopen(
+  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 846, in urlopen
+    return self.urlopen(
+  [Previous line repeated 7 more times]
+  File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 836, in urlopen
+    retries = retries.increment(method, url, response=response, _pool=self)
+  File "/usr/lib/python3.8/site-packages/urllib3/util/retry.py", line 573, in increment
+    raise MaxRetryError(_pool, url, error or ResponseError(cause))
+urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='cray-dhcp-kea-api', port=8000): Max retries exceeded with url: / (Caused by ResponseError('too many 503 error responses'))
+
+During handling of the above exception, another exception occurred:
+
+Traceback (most recent call last):
+  File "/srv/unbound/manager.py", line 573, in <module>
+    main()
+  File "/srv/unbound/manager.py", line 167, in main
+    resp = kea_api('POST', '/', headers=kea_headers, json=kea_request)
+  File "/srv/unbound/manager.py", line 83, in __call__
+    response = http.request(method=method, url=url, headers=headers, **kwargs)
+  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 542, in request
+    resp = self.send(prep, **send_kwargs)
+  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 655, in send
+    r = adapter.send(request, **kwargs)
+  File "/usr/lib/python3.8/site-packages/requests/adapters.py", line 507, in send
+    raise RetryError(e, request=request)
+requests.exceptions.RetryError: HTTPConnectionPool(host='cray-dhcp-kea-api', port=8000): Max retries exceeded with url: / (Caused by ResponseError('too many 503 error responses'))
+```
+Looking for the Kea we noticed that is also in bad state, it is reporting "failed to select a subnet...." messages meaning it hasn't loaded the config.
```

Comment:
3. Check the Kea logs for more information about why it is in a bad state.

    ```sh
    ncn-m001# kubectl -n services logs cray-dhcp-kea-84b4dd8c5d-ng4mc -c cray-dhcp-kea | head
    2022-04-19 13:29:25.948 ERROR [kea-dhcp4.bad-packets/14.139702139406208] DHCP4_PACKET_NAK_0001 [hwtype=1 00:40:a6:83:51:a6], cid=[ff:a6:83:51:a6:00:01:00:01:29:ea:f0:a5:00:40:a6:83:51:a6], tid=0x4cef7c4: failed to select a subnet for incoming packet, src 10.2.0.5, type DHCPDISCOVER
    2022-04-19 13:29:25.999 ERROR [kea-dhcp4.bad-packets/14.139702139406208] DHCP4_PACKET_NAK_0001 [hwtype=1 00:40:a6:83:51:48], cid=[ff:a6:83:51:48:00:01:00:01:29:ea:f0:76:00:40:a6:83:51:48], tid=0x4ec73527: failed to select a subnet for incoming packet, src 10.2.0.5, type DHCPDISCOVER
    2022-04-19 13:29:26.023 ERROR [kea-dhcp4.bad-packets/14.139702139406208] DHCP4_PACKET_NAK_0001 [hwtype=1 00:40:a6:83:02:8b], cid=[ff:a6:83:02:8b:00:01:00:01:29:ea:f0:65:00:40:a6:83:02:8b], tid=0x18dba25f: failed to select a subnet for incoming packet, src 10.2.0.5, type DHCPDISCOVER
    ```

    Kea is getting a 503 error when trying to communicate with the System Layout Service (SLS) so it cannot retrieve the list of networks to build the configuration.

    ```sh
    / $ /srv/kea/dhcp-helper.py
    ERROR: 503 Server Error: Service Unavailable for url: http://cray-sls/v1/search/hardware?type=comptype_cabinet
    ```